nixos/modules/services/security/nwaku.nix

```diff
@@ -0,0 +1,318 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib)
+    types mkEnableOption mkOption mkIf forEach
+    escapeShellArg mdDoc literalExpression
+    optionals concatStringsSep stringLength boolToString;
+
+  cfg = config.services.nwaku;
+in {
+  options = {
+    services = {
+      nwaku = {
+        enable = mkEnableOption (mdDoc "Nim Waku node service.");
+
+        package = mkOption {
+          type = types.package;
+          default = pkgs.nwaku;
+          defaultText = literalExpression "pkgs.nwaku";
+          description = mdDoc "Package to use as Nim Waku node.";
+        };
+
+        protocols = mkOption {
+          type = types.listOf types.str;
+          default = ["relay" "filter" "swap" "lightpush"];
+          example = ["relay" "store" "filter" "swap" "lightpush" "rln-relay" "relay-peer-exchange"];
+          description = mdDoc "List of protocols to enable on the node.";
+        };
+
+        nodekey = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234";
+          description = mdDoc "P2P node private key as 64 char hex string.";
+        };
+
+        storeMessage = {
+          db = {
+            url = mkOption {
+              type = types.str;
+              default = "sqlite://var/run/nwaku/store.sqlite3";
+              description = mdDoc "The database URL for persistent storage.";
+            };
+
+            vacuum = mkOption {
+              type = types.bool;
+              default = false;
+              description = mdDoc "Enable database vacuuming at start. Only supported by SQLite.";
+            };
+
+            migration = mkOption {
+              type = types.bool;
+              default = true;
+              description = mdDoc "Enable database migration at start.";
+            };
+          };
+
+          retentionPolicy = mkOption {
+            type = types.str;
+            default = "time:172800"; # 2 days
```

Comment:
I assume seconds?

Reply:
Do you need to assume?
https://github.com/NixOS/nixpkgs/blob/7dd2d027b4cadf8c829b446c76171c64c4e8b4a1/nixos/modules/services/security/nwaku.nix#L60-L64